starport/cmd/network_campaign_account.go

```diff
@@ -0,0 +1,139 @@
+package starportcmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/tendermint/starport/starport/pkg/entrywriter"
+	"github.com/tendermint/starport/starport/services/network"
+	"github.com/tendermint/starport/starport/services/network/networktypes"
+)
+
+var (
+	campaignMainnetsAccSummaryHeader = []string{"Mainnet Account", "Shares"}
+	campaignVestingAccSummaryHeader  = []string{"Vesting Account", "Total Shares", "Vesting", "End Time"}
+)
+
+// NewNetworkCampaignAccount creates a new campaign account command that holds some other
+// sub commands related to account for a campaign.
+func NewNetworkCampaignAccount() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "account",
+		Short: "Handle campaign accounts",
+	}
+
+	c.AddCommand(
+		newNetworkCampaignAccountList(),
+	)
+	c.PersistentFlags().AddFlagSet(flagNetworkFrom())
+	c.PersistentFlags().AddFlagSet(flagSetKeyringBackend())
+
+	return c
+}
+
+func newNetworkCampaignAccountList() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "list [campaign-id]",
+		Short: "Show all mainnet and mainnet vesting of the campaign",
+		Args:  cobra.ExactArgs(1),
+		RunE:  newNetworkCampaignAccountListHandler,
+	}
+	return c
+}
+
+func newNetworkCampaignAccountListHandler(cmd *cobra.Command, args []string) error {
+	nb, campaignID, err := networkChainLaunch(cmd, args)
+	if err != nil {
+		return err
+	}
+	defer nb.Cleanup()
+
+	n, err := nb.Network()
+	if err != nil {
+		return err
+	}
+
+	accountSummary := &bytes.Buffer{}
+
+	// get all campaign accounts
+	mainnetAccs, vestingAccs, err := getAccounts(cmd.Context(), n, campaignID)
+	if err != nil {
+		return err
+	}
+
+	mainnetAccEntries := make([][]string, 0)
+	for _, acc := range mainnetAccs {
+		mainnetAccEntries = append(mainnetAccEntries, []string{
+			acc.Address,
+			acc.Shares.String(),
+		})
+	}
+	if len(mainnetAccEntries) > 0 {
+		if err = entrywriter.MustWrite(
+			accountSummary,
+			campaignMainnetsAccSummaryHeader,
+			mainnetAccEntries...,
+		); err != nil {
+			return err
+		}
+	}
+
+	mainnetVestingAccEntries := make([][]string, 0)
+	for _, acc := range vestingAccs {
+		mainnetVestingAccEntries = append(mainnetVestingAccEntries, []string{
+			acc.Address,
+			acc.TotalShares.String(),
+			acc.Vesting.String(),
+			strconv.FormatInt(acc.EndTime, 10),
+		})
+	}
+	if len(mainnetVestingAccEntries) > 0 {
+		if err = entrywriter.MustWrite(
+			accountSummary,
+			campaignVestingAccSummaryHeader,
+			mainnetVestingAccEntries...,
+		); err != nil {
+			return err
+		}
+	}
+
+	nb.Spinner.Stop()
+	fmt.Print(accountSummary.String())
```

Comment:
If there are no accounts, can we print an information about it with the warning (i) bullet. Otherwise it feels like command is broken.